Mathlib/Tactic/NormNum/Mod.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2023 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+import Mathlib.Tactic.NormNum.Basic
+import Mathlib.Tactic.NormNum.Ineq
+
+/-!
+# `norm_num` extension for modulo operator
+
+This file adds support for the `· % ·` operator to the `norm_num` tactic.
+-/
+
+set_option autoImplicit true
+
+namespace Mathlib
+open Lean hiding Rat mkRat
+open Meta
+
+namespace Meta.NormNum
+open Qq
+
+lemma isInt_emod {a b q r m : ℤ} (hm : q * b = m) (h : r + m = a) (h₁ : 0 ≤ r) (h₂ : r < b) :
+    IsInt (a % b) r :=
+  ⟨by rw [← h, ← hm, Int.add_mul_emod_self, Int.emod_eq_of_lt h₁ h₂, Int.cast_id]⟩
+
+lemma isInt_emod_neg {a b c : ℤ} (h : IsInt (a % -b) c) : IsInt (a % b) c :=
+  ⟨(Int.emod_neg _ _).symm.trans h.out⟩
+
+lemma isInt_refl (a : ℤ) : IsInt a a :=
+  ⟨rfl⟩
+
+lemma isNat_neg_of_isNegNat {a : ℤ} {b : ℕ} (h : IsInt a (- b)) :
+    IsNat (-a) b :=
+  ⟨by simp [h.out]⟩
+
+/-- Given expressions `a b` in `ℤ`, evaluate `a % b`.
+
+`a` and `b` should be numeric expressions recognized by `norm_num`.
+-/
+partial def evalIntMod.go (a b : Q(ℤ)) : MetaM (NormNum.Result q($a % $b)) := do
+  let ra ← derive (u := Level.zero) a
+  let rb ← derive (u := Level.zero) b
+  let rec core : (b : Q(ℤ)) → Result b → MetaM (Result (u := Level.zero) (α := q(ℤ)) q($a % $b)) :=
```

Comment:
This can be fixed, cf. https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/doc.20lint.20on.20.60let.20rec.60.3F/near/390376885

Reply:
Thanks!